webapp/src/components/App/Studies/StudyTree/index.tsx

```diff
@@ -0,0 +1,184 @@
+    // The thing is at the first render studiesTree was empty.
+    // This made updateTree override studiesTree with an empty tree during the first call. This caused a bug where we didn't see the default
+    // workspace in the UI, as it was overridden by an empty tree at start and then the get workspaces api never returns the default workspace.
+    // Now we don't use the closure anymore, we pass the root tree as a parameter. Thus at the first call of updateTree, we pass initialStudiesTree.
+    // You may think why we don't just capture initialStudiesTree then, it's because in the following call we need to pass studiesTree.
+    let treeAfterWorkspacesUpdate = rootNode;
+
+    let pathsToFetch: string[] = [];
+    // If the user clicks on the root folder, we fetch the workspaces and insert them.
+    // Then we fetch the direct subfolders of the workspaces.
+    if (itemId === "root") {
+      try {
+        treeAfterWorkspacesUpdate = await fetchAndInsertWorkspaces(rootNode);
+        pathsToFetch = treeAfterWorkspacesUpdate.children
+          .filter((t) => t.name !== "default") // We don't fetch the default workspace subfolders, api don't allow it
+          .map((child) => `root${child.path}`);
+      } catch (error) {
+        enqueueErrorSnackbar(
+          "studies.tree.error.failToFetchWorkspace",
+          toError(error),
+        );
+      }
+    } else {
+      // If the user clicks on a folder, we add the path of the clicked folder to the list of paths to fetch.
+      pathsToFetch = [`root${selectedNode.path}`];
+    }
+    const [treeAfterSubfoldersUpdate, failedPath] =
+      await fetchAndInsertSubfolders(pathsToFetch, treeAfterWorkspacesUpdate);
+    if (failedPath.length > 0) {
+      enqueueErrorSnackbar(
+        t("studies.tree.error.failToFetchFolder", {
+          path: failedPath.join(" "),
+          interpolation: { escapeValue: false },
+        }),
+        t("studies.tree.error.detailsInConsole"),
+      );
+    }
+    setStudiesTree(treeAfterSubfoldersUpdate);
+  }
+
+  ////////////////////////////////////////////////////////////////
+  // Event Handlers
+  ////////////////////////////////////////////////////////////////
+
+  const handleTreeItemClick = async (
+    itemId: string,
+    studyTreeNode: StudyTreeNode,
+  ) => {
+    dispatch(updateStudyFilters({ folder: itemId }));
+    updateTree(itemId, studiesTree, studyTreeNode);
+  };
+
+  ////////////////////////////////////////////////////////////////
+  // JSX
+  ////////////////////////////////////////////////////////////////
+
+  const buildTree = (children: StudyTreeNode[], parentId?: string) => {
+    return children.map((child) => {
+      const shouldDisplayLoading =
+        child.hasChildren && child.children.length === 0;
+      const id = parentId ? `${parentId}/${child.name}` : child.name;
+
+      if (shouldDisplayLoading) {
```

Comment:
nothing wrong about this but generally prefer using prefix is, like "isDoingSomething" for flags (the common React boolean flag naming convention), here `isLoadingFolder` or `isLoading` seems better